src/utils/heap.zig

```diff
@@ -1,10 +1,15 @@
 const std = @import("std");
 const Allocator = std.mem.Allocator;
+const builtin = @import("builtin");
 
 const main = @import("main");
 
 var testingErrorHandlingAllocator = ErrorHandlingAllocator.init(std.testing.allocator);
-pub var testingAllocator = testingErrorHandlingAllocator.allocator();
+pub var testingAllocator = blk: {
+	if(!builtin.is_test) @compileError("testing allocator used when not testing");
+
+	break :blk testingErrorHandlingAllocator.allocator();
+};
```

Comment:
This code already exists for std.testing.allocator, why do we need it here again?